### PR TITLE
[test] cover python 3.7 to 3.9 on CPU

### PR DESCRIPTION
**Pull request by min-xu-ai**
- covering common python versions on CPU

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by min-xu-ai** [comment]
python 3.9 seems to have trouble running the unit tests. The problem seems to be random segv after a subprocess finishes. The tests seems to run fine but process exit logic seems to be broken. I think for now we can at least make sure linting all works on py3.9 and re-enable the tests when py3.9 is more stable.

**Comment by blefaudeux** [comment]
I've another branch up to restore torch 1.5 and 1.6 testing (surprise, a lot of things were broken since we were not testing.. ). Would you mind waiting for that to be done before rebasing this one  @min-xu-ai  ? cc @msbaines

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,6 +80,17 @@ install_dep_17_cpu: &install_dep_17_cpu
         python -c 'import torch; print("Torch version:", torch.__version__)'
         python -m torch.utils.collect_env
 
+install_dep_17_cpu_cu11: &install_dep_17_cpu_cu11
+  - run:
+      name: Install Dependencies
+      command: |
+        sudo apt-get install -y libopenmpi-dev
+        pip install --progress-bar off torch==1.7.1+cu110 -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (blefaudeux) [inline .circleci/config.yml:88]: I'm not seeing where cuda 11 is actually requested on the agent, bar that this will run on cpu only I think

Comment (blefaudeux) [reply .circleci/config.yml:88]: given the other PR (thanks for waiting and sorry for the merge conflicts !) I think you'll probably get something good with just installing requirements if you want 1.7 on cpu, no need to install a specific version, at least that's my guess

Comment (min-xu-ai) [reply .circleci/config.yml:88]: No problem. Thanks for fixing the other issue. That will make this PR even better. I will handle the conflicts.

Yeah for cpu, we can just install the version without cuda. I will test it out.

Comment (msbaines) [reply .circleci/config.yml:88]: To support CUDA 11.0, we would need to create a docker image with the correct kernel driver.

**Comment by min-xu-ai** [comment]
Hi @msbaines @blefaudeux, I fixed up this PR. We now have two more things:

- we cover py 3.7 to 3.9, in terms of static checking for py39. Py39 is not yet stable to run all unit tests.
- we also cover doc build so that doc problems can be detected earlier in the PRs.

What do you guys think?

File: .circleci/config.yml [hunk 2]
```diff
@@ -135,20 +165,30 @@ run_oss_gloo: &run_oss_gloo
...
-    name: Run OSS with Torch AMP
-    command: |
-      python benchmarks/oss.py --amp --epochs 3 --optim_type oss_sharded_ddp
+   - run:
+       name: Run OSS with Torch AMP
+       command: |
+         python benchmarks/oss.py --amp --epochs 3 --optim_type oss_sharded_ddp
+
+
+run_doc_build: &run_doc_build
```
Comment (blefaudeux) [inline .circleci/config.yml:175]: nice !

File: requirements-test.txt [hunk 3]
```diff
@@ -1,15 +1,16 @@
+# Tools for static checking.
 black == 19.10b0
 flake8 == 3.7.9
 isort == 5.6.4
-mpi4py == 3.0.3
-mypy == 0.770
+mypy == 0.790
```
Comment (blefaudeux) [inline requirements-test.txt:5]: does that change anything, practically ?

**Review by blefaudeux — APPROVED** [review]
lgtm, thanks for this work Min !

File: requirements-test.txt (hunk 3 above)
Comment (min-xu-ai) [reply requirements-test.txt:5]: only mypy version is updated so that py39 can work. no impact on py37 and py38.

Comment (min-xu-ai) [reply requirements-test.txt:5]: mpi4py is moved into a group only.

**Pull request merged**